Contributing information

andreugallofre opened the pull request:
**What this PR does / why we need it**:

Change CONTRIBUTING.md to match the actual information on the repository on the lables used for issues and pull request.

**Special notes for your reviewer**:

I'm not sure that the merging conditions of the tags size/L , size/XL and size/XXL are correct because did not found any documentation about it.

This PR closes #5795

**If applicable**:
- [x] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

bacongobbler commented on CONTRIBUTING.md:
```diff
@@ -327,6 +327,9 @@ lines is greater than defined below.
-| `size/small` | Anything less than or equal to 4 files and 150 lines. Only small amounts of manual testing may be required |
-| `size/medium` | Anything greater than `size/small` and less than or equal to 8 files and 300 lines. Manual validation should be required. |
-| `size/large` | Anything greater than `size/medium`. This should be thoroughly tested before merging and always requires 2 approvals. This also should be applied to anything that is a significant logic change. |
+| `size/XS` | Anything less than or equal to 9 lines ignoring generated files. Only small amounts of manual testing may be required |
```
missing period at end of sentence

bacongobbler commented on CONTRIBUTING.md:
```diff
@@ -327,6 +327,9 @@ lines is greater than defined below.
-| `size/medium` | Anything greater than `size/small` and less than or equal to 8 files and 300 lines. Manual validation should be required. |
-| `size/large` | Anything greater than `size/medium`. This should be thoroughly tested before merging and always requires 2 approvals. This also should be applied to anything that is a significant logic change. |
+| `size/XS` | Anything less than or equal to 9 lines ignoring generated files. Only small amounts of manual testing may be required |
+| `size/S` | Anything greater than `size/XS` less than or equal to 29 lines ignoring the generated files. Only small amounts of manual testing may be required |
```
missing period at end of sentence

bacongobbler commented on CONTRIBUTING.md:
```diff
@@ -327,6 +327,9 @@ lines is greater than defined below.
-| `size/large` | Anything greater than `size/medium`. This should be thoroughly tested before merging and always requires 2 approvals. This also should be applied to anything that is a significant logic change. |
+| `size/XS` | Anything less than or equal to 9 lines ignoring generated files. Only small amounts of manual testing may be required |
+| `size/S` | Anything greater than `size/XS` less than or equal to 29 lines ignoring the generated files. Only small amounts of manual testing may be required |
+| `size/M` | Anything greater than `size/S` less than or equal to 99 lines ignoring the generated files. This should be thoroughly tested before merging and always requires 1 approvals. Manual validation should be required. |
```
Can you remove "This should be thoroughly tested before merging and always requires 1 approvals"? I feel that the second line (manual validation should be required) covers the first statement, and every PR requires at least 1 approval so its's moot to call that out here.

bacongobbler commented on CONTRIBUTING.md:
```diff
@@ -327,6 +327,9 @@ lines is greater than defined below.
+| `size/XS` | Anything less than or equal to 9 lines ignoring generated files. Only small amounts of manual testing may be required |
+| `size/S` | Anything greater than `size/XS` less than or equal to 29 lines ignoring the generated files. Only small amounts of manual testing may be required |
+| `size/M` | Anything greater than `size/S` less than or equal to 99 lines ignoring the generated files. This should be thoroughly tested before merging and always requires 1 approvals. Manual validation should be required. |
+| `size/L` | Anything greater than `size/M` less than or equal to 499 lines ignoring the generated files. This should be thoroughly tested before merging and always requires 1 approvals. This also should be applied to anything that is a significant logic change. |
```
size/L requires 2 approvals.

atsuky commented:
@bacongobbler The commits should be squashed? or can i leave it like that in order to have the complete history? 

Thanks!

bacongobbler commented:
Generally speaking it's better to keep them squashed. Thanks!

atsuky commented:
Done, commits squased! 👍

andreugallofre committed:
Change the starter label for the actual one used

Signed-off-by: Andreu Gallofre <andreugallofre@gmail.com>

Change the label sizing in CONTRIBUTING.md

Change the label size and merging conditions to match the actual labels
and conditions used

Signed-off-by: Andreu Gallofre <andreugallofre@gmail.com>

Add requested changes

Add missing dot, removes duplicated information and change requirements
for merging size/L

Signed-off-by: Andreu Gallofre <andreugallofre@gmail.com>
